components/pedigree/resources/src/main/resources/pedigree/view/nodeMenu.js

```diff
@@ -957,6 +844,112 @@ define([
             }
         },
 
+        _addNewCancerRow: function(cancerId, label, table, disableDelete, toObserve) {
+            var dataName = "cancers";
+            var tableRow = new Element('tr', {'class': 'cancer_field'} );
+            var qualifiers = this._buildNewCancerElement(cancerId, label, dataName,
+              {'allowMultiDialogues': true, 'disableTermDelete': disableDelete});
+            this._setCancersValueFx(qualifiers);
+            this._attachFieldEventListeners(qualifiers, toObserve);
+            table.insert(tableRow.insert(qualifiers));
+            return qualifiers;
+        },
+
+        _setCancersValueFx: function(cancerElem) {
+            var _this = this;
+            cancerElem._getValue = function() {
+                var data = [];
+                _this.form.select('table.cancers-summary-group').each(function(qualifier) {
+                    var cancerWidget = qualifier._widget;
+                    cancerWidget.isAffected() && data.push(cancerWidget.getValues());
+                });
+                return [ data ];
+            };
+        },
+
+        _buildNewCancerElement: function(cancerId, cancerName, dataName, optionsParam) {
+            var toStoredLateralityFx = function(value) {
+                switch(value) {
```

Comment:
I think a map `{"Bilateral": "bi", ... }` is more concise and easier to update, but this is not important

Reply:
Fixed.